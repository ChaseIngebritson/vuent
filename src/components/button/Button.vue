<template>
  <button class="vnt-button"
          :type="type"
          :disabled="disabled"
          @click="click">{{ label }}</button>
</template>

<script>
export default {
  name: 'VntButton',

  props: {
    label: {
      type: String,
      default: 'Button'
    },
    type: {
      type: String,
      default: 'button'
    },
    disabled: {
      type: Boolean,
      default: false
    },
    click: {
      type: Function,
      default: () => {}
    }
  }
};
</script>

<style lang="scss">
@import '../../scss/mixins/component';

.vnt-button {
  @include component-base();

  line-height: 30px;
  background: #ccc;
  padding: 0 37px;
  border: 1px solid #ccc;
  color: #000100;

  &:hover {
    border-color: #979797;
  }

  &:active {
    background: #979797;
  }

  &:disabled {
    color: #7a7a7a;

    &:hover {
      border-color: #ccc;
    }
  }
}
</style>
