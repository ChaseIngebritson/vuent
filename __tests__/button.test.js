import { mount } from 'vue-test-utils';
import Button from '@/components/button';

describe('Button', () => {
  let wrapper;

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(Button);
    });

    test('has correct type', () => {
      expect(wrapper.vm.type).toBe('button');
    });

    test('is not disabled', () => {
      expect(wrapper.vm.disabled).toBe(false);
    });

    test('label text is "Button"', () => {
      expect(wrapper.vm.label).toBe('Button');
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can be disabled', () => {

    beforeAll(() => {
      wrapper = mount(Button, {
        propsData: {
          disabled: true
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    test('is disabled', () => {
      expect(wrapper.vm.disabled).toBe(true);
    });

  });

});
