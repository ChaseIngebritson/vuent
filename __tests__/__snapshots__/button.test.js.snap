// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button by default renders correctly 1`] = `<button type="button" class="vnt-button">Button</button>`;

exports[`Button can be disabled renders correctly 1`] = `<button type="button" disabled="disabled" class="vnt-button">Button</button>`;
